ui/page/dexclient/markets.go

```diff
@@ -0,0 +1,212 @@
+package dexclient
+
+import (
+	"context"
+	"fmt"
+
+	"decred.org/dcrdex/client/asset/btc"
+	"decred.org/dcrdex/client/asset/dcr"
+	"decred.org/dcrdex/client/core"
+	"gioui.org/layout"
+
+	"github.com/planetdecred/godcr/dexc"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+type (
+	C = layout.Context
+	D = layout.Dimensions
+)
+
+const MarketPageID = "Markets"
+
+const testDexHost = "dex-test.ssgen.io:7232"
+
+type selectedMaket struct {
+	host          string
+	name          string
+	marketBase    string
+	marketQuote   string
+	marketBaseID  uint32
+	marketQuoteID uint32
+}
+
+type Page struct {
+	*load.Load
+	ctx             context.Context // page context
+	ctxCancel       context.CancelFunc
+	initializeModal bool
+	addBTCWallet    decredmaterial.Button
+	advancedTrade   decredmaterial.Button
+	selectedMaket   *selectedMaket
+}
+
+// TODO: Aadd collapsible button to select a market.
+var mkt = &selectedMaket{
+	host:          testDexHost,
+	name:          "DCR-BTC",
+	marketBase:    "dcr",
+	marketBaseID:  dcr.BipID,
+	marketQuote:   "btc",
+	marketQuoteID: btc.BipID,
+}
+
+func NewMarketPage(l *load.Load) *Page {
+	pg := &Page{
+		Load:            l,
+		initializeModal: false,
+		selectedMaket:   mkt,
+	}
+
+	return pg
+}
+
+func (pg *Page) ID() string {
+	return MarketPageID
+}
+
+func (pg *Page) OnResume() {
+	pg.ctx, pg.ctxCancel = context.WithCancel(context.TODO())
+	pg.refreshUser()
+
+	if pg.Dexc.Core.User().Initialized && pg.Dexc.IsLoggedIn {
+		go pg.readNotifications()
+	}
+}
+
+func (pg *Page) Layout(gtx C) D {
+	return pg.registrationStatusLayout(gtx)
+}
+
+func (pg *Page) registrationStatusLayout(gtx C) D {
+	dex := pg.Dexc.Core.User().Exchanges[pg.selectedMaket.host]
+	if dex == nil || !dex.Connected {
```

Comment:
Instead of showing status for the dextest host only, this can work with any dex the user connects, in case a user connects a different dex.

For e.g., the code below can pick the first dex on the exchanges map to use, if the map is not empty.

```suggestion
	var dex *core.Exchange
	for _, dex = range pg.Dexc.Core.User().Exchanges {
		break
	}
	if dex == nil || !dex.Connected {
```